src/ui/Facet/Facet.ts

```diff
@@ -1285,11 +1286,34 @@ export class Facet extends Component {
   protected facetValueHasChanged() {
     this.updateQueryStateModel();
     this.rebuildValueElements();
+    this.clearDependentFacetsIfNoSelectedValuesRemain();
     Defer.defer(() => {
       this.updateAppearanceDependingOnState();
     });
   }
 
+  private clearDependentFacetsIfNoSelectedValuesRemain() {
+    if (this.getSelectedValues().length) {
```

Comment:
I would change this to `this.values.hasSelectedOrExcludedValues()`

Reply:
Okay, just to make sure, in the logic for controlling the visibility of the dependent facet, we were only checking if the parent had selected values. I will change it so the dependent facet is visible on selected and excluded values.